### ISSUE TITLE
Fature: Check for unintended non-printing characters

### ISSUE THREAD

**Issue by jaraco** [astral-sh/ruff#13297]
In https://github.com/jaraco/zipp/pull/125, I learned that users can, intentionally or not, sneak non-printable characters into the code through otherwise innocuous pull requests. This particular one was caught by `mach lint` in a downstream system. It would be nice if ruff could catch such changes. I'm not confident it's possible, but I wanted to flag it as a feature for consideration.

**Comment by MichaReiser** [comment]
I'm rather surprised that we don't already have that. There's an existing rule to catch [ambiguous unicode characters](https://docs.astral.sh/ruff/rules/ambiguous-unicode-character-docstring/), but it doesn't cover invisible characters. 

I'm very much in favor of adding the rule. We would have to think about if it should be a new rule or be added to the above mentioned rule

**Comment by xyb** [comment]
I suggest add `--show-all` option also to show non-printable characters like [bat](https://github.com/sharkdp/bat):

![Screenshot 2024-09-11 at 17 26 12](https://github.com/user-attachments/assets/0442d1ee-a24f-4204-a5f6-874e90d877dd)

**Comment by MichaReiser** [comment]
@xyb, this issue is different. It isn't about showing invisible characters; it's about linting for invisible characters.

**Comment by xyb** [comment]
@MichaReiser I agree this is a separate request, but it's related because it would be confusing if Ruff reports an invisible character without helping the user fix it.  If you agree, I'll go ahead and create a new issue to track this requirement.
